src/osd/OSD.cc

```diff
@@ -2403,13 +2403,29 @@ OSD::OSD(CephContext *cct_,
   trace_endpoint.copy_name(ss.str());
 #endif
 
+  // Determine scheduler type for this OSD
+  auto get_scheduler_type = [&conf = cct->_conf]() {
+    if (auto type = conf.get_val<std::string>("osd_op_queue");
+        type != "debug_random") {
+      return type;
+    } else {
+      static const std::vector<std::string> index_lookup = { "mclock_scheduler",
+                                                             "wpq" };
+      std::mt19937 random_gen(std::random_device{}());
+      auto which = random_gen() % index_lookup.size();
+      return index_lookup[which];
+    }
+  };
+  std::string osd_op_queue = get_scheduler_type();
```

Comment:
This should return the enum class mentioned above.

Reply:
Done. Note: I changed the name to get_op_queue_type().

Reply:
Note: I probably need to perform as assertion check that the op_queue returned is not "unknown".